Refactors rust lint out into its own job which runs on a bare GH runner

ibeckermayer opened the pull request:
Also adds -s to silence all make commands in workflows which are used as some form of output in the workflow script. This is to prevent the workflow from failing due to noise from the make command (such as when make tells you it's entering a directory, etc).

Also updates to latest e-ref

Continuation of https://github.com/gravitational/teleport/pull/40021

Commit:
Refactors rust lint out into its own job which runs on a bare GH runner

Also adds -s to silence all make commands in workflows which are used as
some form of output in the workflow script. This is to prevent the
workflow from failing due to noise from the make command (such as when
make tells you it's entering a directory, etc).

Also updates to latest e-ref

camscale commented on .github/workflows/build-macos.yaml:
```diff
@@ -35,10 +35,10 @@ jobs:
-          echo GOLANG_VERSION=$(make -C build.assets print-go-version | sed 's/^go//') >> $GITHUB_ENV
-          echo RUST_VERSION=$(make -C build.assets print-rust-version) >> $GITHUB_ENV
-          echo WASM_PACK_VERSION=$(make -C build.assets print-wasm-pack-version) >> $GITHUB_ENV
+          echo NODE_VERSION=$(make -s -C build.assets print-node-version) >> $GITHUB_ENV
```
You should also add `--no-print-directory` like it has in the build-windows workflow if you want to stop the "Entering directory..." messages.

ibeckermayer replied:
It looks like `-s` includes `--no-print-directory` by default:

> As an aside, notice that the --no-print-directory option is automatically activated if the -s flag is used.

https://www.gnu.org/software/automake/manual/html_node/Tricks-For-Silencing-Make.html

camscale replied:
Good to know. It would be nice if the man page said that too, but I know how much GNU dislike man pages and prefer their info format instead. But even the info document index does not have either `-s` or `--no-print-directory`.

camscale commented on Makefile:
```diff
@@ -1174,44 +1174,30 @@ update-tag:
 	(cd e && git tag $(GITTAG) && git push origin $(GITTAG))
 	git push $(TAG_REMOTE) $(GITTAG) && git push $(TAG_REMOTE) api/$(GITTAG)
 
-# HAS_CLOUD_SEMVER is non-empty if $(VERSION) contains a cloud-only pre-release tag,
```
For some reason this is reverting changes I made recently for cloud-only releases.

ibeckermayer replied:
My bad, I was trying to do some git fu to squash this to a single commit and goofed something. Should be resolved now.

zmb3 reviewed (approved):
LGTM assuming you re-introduce Cam's changes.

Is the teleport.e update necessary here? This will fail auto backports with an e update, likely creating more work for you.

Commit:
Removes `--no-print-directory` where `make` output is used as a variable
replaceing it with `-s` instead. `-s` is a universal POSIX standard and
makes sense in all such cases.

https://www.gnu.org/software/automake/manual/html_node/Tricks-For-Silencing-Make.html

Commit:
aligning e with master

Commit:
aligning e with master

Commit:
aligning e with master

Commit:
Merge branch 'master' into isaiah/lint-rust-2
